front-end/src/App.js

```diff
@@ -1,21 +1,54 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import 'tachyons';
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
-  }
+	constructor() {
+		super()
+		this.state = {
+			songUrl: ''
+		}
+	}
+
+	performSearch = () => {
+		console.log("Searching...")
+	}
+
+	onSkipPressed = () => {
+		console.log('Skip pressed')
+	}
+
+	onPlayPressed = () => {
+		console.log(this.state.songUrl)
+	}
+
+	onStopPressed = () => {
+		console.log('Stop pressed')
+	}
+
+	onUrlChanged = (event) => {
+		this.setState({
+			songUrl: event.target.value
+		})
+	}
+
+	render() {
+		return (
+			<div className="App">
+				<div className="pt3 measure center">
+					<div className="ma2">
+						<input type="text" name="songUrl" onChange={this.onUrlChanged} />
+						<button id="search" onClick={this.performSearch} className="mt1 ml2"> Search </button>
+					</div>
+					<div className="mt2 pl2 pr2">
+						<button id="skip" onClick={this.onSkipPressed} className="mr3 ml3"> skip </button>
+						<button id="play" onClick={this.onPlayPressed} className="mr3 ml3"> play </button>
+						<button id="stop" onClick={this.onStopPressed} className="mr3 ml3"> stop </button>
+					</div>
+				</div>
+			</div>
+		);
+	}
 }
 
 export default App;
```

Comment:
Exporting the module can be done in the first line itself if you don't have to perform any functions with it. Else if you want to perform any functions with the module you are exporting ( like wrapping it in a function ) this is the appropriate position.